pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -993,17 +1020,42 @@ class Timedelta(_Timedelta):
         return self.value // other.value
 
     def __rfloordiv__(self, other):
+        # numpy does not implement floordiv for timedelta64 dtype, so we cannot
+        # just defer
+        if hasattr(other, '_typ'):
+            # Series, DataFrame, ...
+            return NotImplemented
+
         if hasattr(other, 'dtype'):
-            # work with i8
-            other = other.astype('m8[ns]').astype('i8')
-            return other // self.value
+            if other.dtype.kind == 'm':
+                # also timedelta-like
+                # We need to watch out for np.timedelta64('NaT')
+                if np.ndim(other) == 0:
+                    if np.isnat(other):
+                        return np.nan
+                    else:
+                        other = other.astype('m8[ns]').astype('i8')
+                        return other // self.value
+                else:
+                    mask = np.isnat(other)
+                    res = other.astype('m8[ns]').astype('i8') // self.value
+                    if mask.any():
+                        res = res.astype('f8')
+                        res[mask] = np.nan
+                    return res
+            else:
+                raise TypeError('Invalid dtype {dtype} for '
+                                '{op}'.format(dtype=other.dtype,
+                                              op='__floordiv__'))
 
+        if isinstance(other, float) and np.isnan(other):
+            return NotImplemented
         elif not _validate_ops_compat(other):
             return NotImplemented
 
         other = Timedelta(other)
         if other is NaT:
-            return NaT
+            return np.nan
```

Comment:
is this what we do for division?
is it not tested now?

Reply:
> is this what we do for division?

Yes.

The status quo is... weird.

```
    def __floordiv__(self, other):
        if hasattr(other, 'dtype'):
            # work with i8
            other = other.astype('m8[ns]').astype('i8')
            return self.value // other

        elif is_integer_object(other):
            # integers only
            return Timedelta(self.value // other, unit='ns')

        elif not _validate_ops_compat(other):
            return NotImplemented

        other = Timedelta(other)
        if other is NaT:
            return np.nan
        return self.value // other.value

    def __rfloordiv__(self, other):
        if hasattr(other, 'dtype'):
            # work with i8
            other = other.astype('m8[ns]').astype('i8')
            return other // self.value

        elif not _validate_ops_compat(other):
            return NotImplemented

        other = Timedelta(other)
        if other is NaT:
            return NaT
        return other.value // self.value

```

> is it not tested now?

Not that I'm aware of.